src/EFCore.PG/Query/Internal/NpgsqlQuerySqlGenerator.cs

```diff
@@ -206,6 +207,39 @@ protected override Expression VisitSqlBinary(SqlBinaryExpression binary)
         }
     }
 
+    protected override Expression VisitCommandRootExpression(Expression rootExpression)
```

Comment:
@smitpatel when doing PostgresDeleteExpression, I was blocked by QuerySqlGenerator.GetCommand, where I needed to dispatch to my custom expression; but GetCommand cannot really be overridden (since _relationalCommandBuild isn't writable).

Instead of identifying and dispatching for specific expression types in GetCommand (Select/Delete) and throwing for unknowns, we could simply do Visit, and only handle SelectExpression in a special way there (for the tags, for FromSql); that would obviate having to update GetCommand for new expression types which can appear at the root.

Or if you feel strongly that we should keep the current model, we can refactor the dispatch code to another method (VisitCommandRootExpression?) which can be overridden by providers (that's what I currently did here). But my preference would be to just call Visit in GetCommand.

Reply:
Let me look into it. I don't have strong preference for current way.

Reply:
@smitpatel just a reminder that this is still pending - I can't merge the PG support without this (though that isn't very urgent).

Reply:
https://github.com/dotnet/efcore/pull/28582